src/table/src/engine/manager.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+use crate::engine::TableEngineRef;
+
+pub trait TableEngineManager: Send + Sync {
+    fn engine(&self, name: &str) -> Option<TableEngineRef>;
+    fn register_engine(&self, name: &str, engine: TableEngineRef);
+    fn default(&self) -> TableEngineRef;
+}
+pub type TableEngineManagerRef = Arc<dyn TableEngineManager>;
+
+/// Simple in-memory table engine manager
+pub struct MemoryTableEngineManager {
+    pub engines: RwLock<HashMap<String, TableEngineRef>>,
+    pub default_token: String,
+}
+
+impl MemoryTableEngineManager {
+    pub fn new(default: &str, engine: TableEngineRef) -> Self {
+        let engines = RwLock::new(HashMap::new());
+        let default_token = default.to_string();
+        // it's safe to unwrap
+        engines.write().unwrap().insert(default_token, engine);
+
+        MemoryTableEngineManager {
+            engines,
+            default_token: default.to_string(),
+        }
+    }
+}
+
+impl TableEngineManager for MemoryTableEngineManager {
+    fn engine(&self, name: &str) -> Option<TableEngineRef> {
+        let engines = self.engines.read().unwrap();
+        engines.get(name).cloned()
+    }
+
+    fn register_engine(&self, name: &str, engine: TableEngineRef) {
+        self.engines
+            .write()
+            .unwrap()
+            .insert(name.to_string(), engine);
```

Comment:
Better return some error if the engine with same name is registered. Table engine must be unique for a certain kind.